numba_cuda/numba/cuda/cuda_paths.py

```diff
@@ -256,3 +258,66 @@ def get_debian_pkg_libdevice():
     if not os.path.exists(pkg_libdevice_location):
         return None
     return pkg_libdevice_location
+
+
+def get_current_cuda_target_name():
+    """Determine conda's CTK target folder based on system and machine arch.
+
+    CTK's conda package delivers headers based on its architecture type. For example,
+    `x86_64` machine places header under `$CONDA_PREFIX/targets/x86_64-linux`, and
+    `aarch64` places under `$CONDA_PREFIX/targets/sbsa-linux`. Read more about the
+    nuances at cudart's conda feedstock:
+    https://github.com/conda-forge/cuda-cudart-feedstock/blob/main/recipe/meta.yaml#L8-L11  # noqa: E501
+    """
+    system = platform.system()
+    machine = platform.machine()
+
+    if system == "Linux":
+        arch_to_targets = {
+            'x86_64': 'x86_64-linux',
+            'aarch64': 'sbsa-linux'
+        }
+        return arch_to_targets.get(machine)
+
+    return None
+
+
+def get_conda_include_dir():
+    """
+    Return the include directory in the current conda environment, if one
+    is active and it exists.
+    """
+    conda_prefix = os.environ.get('CONDA_PREFIX')
+    target_name = get_current_cuda_target_name()
+
+    if conda_prefix:
+        if target_name:
+            include_dir = os.path.join(
+                conda_prefix, f'targets/{target_name}/include'
+            )
+        else:
+            # A fallback when target cannot determined
+            # though usually it shouldn't.
+            include_dir = os.path.join(conda_prefix, 'include')
+        if os.path.exists(include_dir):
+            return include_dir
+    return None
+
+
+def get_system_include_dir():
+    """Return the system CUDA include directory, if it exists"""
+    system_cuda_include = '/usr/local/cuda/include'
+    if os.path.exists(system_cuda_include):
+        return system_cuda_include
+    return None
+
+
+def _get_include_dir():
+    """Find the root include directory."""
+    options = [
+        ('Conda environment', get_conda_include_dir()),
+        ('System', get_system_include_dir()),
+        # TODO: add others
```

Comment:
I think we need to search the `config.CUDA_INCLUDE_PATH` before these other two options, otherwise users who relied on setting it will find it no longer has any effect.

Reply:
Note existing logic in https://github.com/numba/numba/blob/main/numba/core/config.py#L467-L478

Reply:
I think there's a dilemma here. The desired solution is to have a mechanism to detect if user has specified `NUMBA_CUDA_INCLUDE_PATH` in configuration, if it's not specified, `config.CUDA_INCLUDE_PATH` returns none and uses other defaults. However, `numba.config` doesn't support such behavior and `config.CUDA_INCLUDE_PATH` always has a value.